compiler/utils/astrepr.nim

```diff
@@ -122,124 +200,175 @@ proc format[I](s: set[I], sub: int): string =
 
   result.add "}"
 
-const setStyle = termFg(2, 1, 3)
+func format(i: SomeInteger): ColText = $i + style.number
+
+proc ownerChain(sym: PSym): string =
+  var own: seq[PSym] = @[sym]
+  var sym = sym
+  while not sym.owner.isNil():
+    sym = sym.owner
+    own.add sym
 
-func format(i: SomeInteger): ColText = $i + fgCyan
+  for idx in countdown(own.high, 0):
+    if idx < own.high:
+      result.add "."
+
+    result.add own[idx].name.s
```

Comment:
Mildly faster and sightly more readable would be a string format, with '%'. I care about readability more, of course.